lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -52,6 +52,7 @@ def as_json(_options)
           client_name: client.name,
           status: I18n.t("doorkeeper.pre_authorization.status"),
         }
+        attributes.is_a?(Hash) ? pre_auth.merge(attributes) : pre_auth
```

Comment:
We can impove it in the way like:

`pre_auth.merge(attributes.to_h) if attributes.respond_to?(:to_h)`

This will allow any object to be passed that can be converted to a hash, WDYT?

Reply:
I can change it to that, sure. However we still need to return the pre_auth if the attributes don't respond to `to_h` right ?

Reply:
Sure! Just a tiny improvement to allow any "hashable" object to be passed :)